generic/piranha-tree-sitter/src/utilities/mod.rs

```diff
@@ -0,0 +1,109 @@
+/*
+Copyright (c) 2022 Uber Technologies, Inc.
+
+ <p>Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file
+ except in compliance with the License. You may obtain a copy of the License at
+ <p>http://www.apache.org/licenses/LICENSE-2.0
+
+ <p>Unless required by applicable law or agreed to in writing, software distributed under the
+ License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ express or implied. See the License for the specific language governing permissions and
+ limitations under the License.
+*/
+
+pub mod tree_sitter_utilities;
+
+use std::collections::HashMap;
+#[cfg(test)]
+use std::fs::{self, DirEntry};
+use std::fs::{File, OpenOptions};
+use std::hash::Hash;
+use std::io::{BufReader, Read};
+use std::path::PathBuf;
+use toml;
+
+// Reads a file.
+pub fn read_file(file_path: &PathBuf) -> Result<String, String> {
```

Comment:
one thing to explore in a followup is to add unit tests for these functions because the refactorings are e2e tests.